[website] Add GitHub contributor activity page

dwang opened the pull request:
A new page is added to display the events that each contributor triggers (stars, pull requests, comments, etc). 
The page is populated using data from the GitHub API and jQuery to parse the JSON.

Preview can be seen at: [openwisp-website.surge.sh/activity.html](http://openwisp-website.surge.sh/activity.html)

Alternatively, here is an image:
![image](https://user-images.githubusercontent.com/34558138/48460990-801cfd80-e79f-11e8-8969-6fa401504639.png)

atb00ker commented:
If XYZ comments on a PR, is it possible to directly jump to that comment or issue/pull request ? 
Personally, i find that very useful, saves the time to find the comment.
what do you think?

dwang commented:
Thanks for reviewing, will work on it now.

dwang commented on activity.html:
```diff
@@ -0,0 +1,237 @@
+      return interval + ' ' + intervalType;
+    };
+
+    function setFooter() {
```
I'm assuming you are asking about the `setFooter()` function. Without it, the footer would be fixed in the middle of the screen, even after the feed was populated. So now the footer is being moved to the bottom of the page.

dwang commented on activity.html:
```diff
@@ -0,0 +1,237 @@
+  <script src="js/scripts.js"></script>
+  <script>
+    function createEvent(avatar, name, event, repo, header, content, date) {
+      if (!event)
```
`Event` can be false because I am passing it to the function as a parameter, and it's the string representation of that event. For example, `event` could be "created a new issue in". It's there because unhandled events would be empty strings and I didn't want to create a new message in the feed for it.

dwang commented:
@atb00ker Just added it, the title in the message box is a link to the corresponding comment or PR.

@nemesisdesign 

- move the JS logic to a separate file
**Moved the logic to `js/scripts.js`**
- try to have a single function that can be called to create the feed
**Didn't really know exactly what you wanted, but I moved the `createEvent()` function into the main function. I also simplified the `timeSince()` function.**
- PullRequestReviewCommentEvent is missing but those kind of events are important
**`PullRequestReviewCommentEvent`, `PushEvent`, and their relevant links/content are added. Let me know if any other events should be accounted for.**
- it's important to show convert the markdown formatting otherwise comments are not readable, use a markdown tool, maybe this one can work: https://github.com/showdownjs/showdown
**Added `showdown` and it looks like it's working correctly. I haven't tested it much, but Markdown text looks alright. I'm unsure about images though and how they would be formatted. The `js/showdown.min.js` file is causing Travis to fail, should I do anything about it? Maybe use a CDN instead, or ignore it in JSHint?**
- we should convert URLs to links with regular expression, we can do this as a last step, try to google existing solutions to this problem (it's a common problem)
**Will start working on this now, just wanted to commit my other changes first, and get feedback when I'm working on this.**

dwang commented:
Moved the logic over to `js/feed.js` now. Still have a few more things I want to do. The URL parsing, a collapse button for the messages, and Markdown image formatting. Anything else I should do?

dwang commented:
Demo is also updated with the current changes: http://openwisp-website.surge.sh/activity.html

dwang commented:
Simple URLs are currently being matched based on the protocol (http:// and https://) and they get converted to clickable links. If needed, I can change it to match based on TLD or other aspects of the URL.

dwang commented:
Found an issue, links break if the URL is split into multiple lines. Don't really know how to fix this?
Example:
<img width="1204" alt="screen shot 2018-11-16 at 11 15 24 am" src="https://user-images.githubusercontent.com/34558138/48633362-21b77100-e991-11e8-8c3e-39a786f57c05.png">

nemesisdesign commented on activity.html:
```diff
@@ -0,0 +1,90 @@
+            <a class="item" href="http://openwisp.io/docs">Docs</a>
+            <a class="item" href="support.html">Support</a>
+            <a class="item" href="thecode.html">Code</a>
+            <a class="item active" href="activity.html">Activity</a>
```
Remove this as well

nemesisdesign commented on activity.html:
```diff
@@ -0,0 +1,90 @@
+    <a class="item" href="http://openwisp.io/docs">Docs</a>
+    <a class="item" href="support.html">Support</a>
+    <a class="item" href="thecode.html">Code</a>
+    <a class="item active" href="activity.html">Activity</a>
```
remove this as well

dwang commented:
@nemesisdesign My editor has been updated and the issues have been resolved. Anything else I should take care of? Thanks!
